src/app/courses/courses.service.ts

```diff
@@ -60,7 +60,7 @@ export class CoursesService {
   }
 
   updateSubmission(answer, index: number, close: boolean) {
-    const submission = { ...this.submission, answers: [ ...this.submission.answers ] };
+    const submission = { ...this.submission, answers: this.submission.answers ? [ ...this.submission.answers ] : [] };
```

Comment:
`answers: [ ...this.submission.answers ]` is not required at all as it will assign answers of submission to answers itself without any modification